### PR TITLE
Refactor publisher, preparing it for switching the eventing backend

### PR DESCRIPTION
**Pull request by radufa**
**Description**
Dynamically switching the eventing backend 

Changes proposed in this pull request:

- Refactor to better reflect the used backend
- Extend the signals package to expose a new multiple instance  context 

**Related issue(s)**

### CODE REVIEW

File: components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go [hunk 1]
```diff
@@ -92,11 +96,17 @@ func (c *Commander) Start() error {
...
 		return err
 	}
 	c.logger.Info("Shutdown the Event Publisher Proxy")
 	return nil
 }
+
+// Stop implements the Commander interface and stops the publisher.
+func (c *Commander) Stop() error {
+	c.cancel()
+	return nil
```
Comment (themue) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go:111]: Don't we need any kind of cleanup here?

**Review by themue — APPROVED** [review]
Only formal comments we should care about in a separate change.

File: components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go (hunk 1 above)
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go:111]: Not for now.

File: components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go [hunk 2]
```diff
@@ -55,14 +58,15 @@ func (c *Commander) Start() error {
...
+	var ctx context.Context
+	ctx, c.cancel = context.WithCancel(signals.NewContext())
 
 	// configure auth client
 	client := oauth.NewClient(ctx, c.envCfg)
 	defer client.CloseIdleConnections()
 
 	// configure message sender
-	messageSender := sender.NewHttpMessageSender(c.envCfg.EmsPublishURL, client)
+	messageSender := sender.NewBebMessageSender(c.envCfg.EmsPublishURL, client)
```
[the remark this answers is not in the document]
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go:69]: Will be solved in: https://github.com/kyma-project/kyma/issues/11254

File: components/event-publisher-proxy/cmd/event-publisher-proxy/nats/nats.go [hunk 3]
```diff
@@ -52,7 +55,8 @@ func (c *Commander) Start() error {
 	c.logger.Infof("Starting Event Publisher to NATS, envCfg: %v; opts: %#v", c.envCfg.String(), c.opts)
```
[the remark this answers is not in the document]
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/nats/nats.go:55]: Created https://github.com/kyma-project/kyma/issues/11255

**Pull request merged**